virtcontainers/clh.go

```diff
@@ -302,19 +284,26 @@ func (clh *cloudHypervisor) startSandbox(timeout int) error {
 	if err != nil {
 		return fmt.Errorf("failed to launch cloud-hypervisor: %s, error messages from log: %s", err, strErr)
 	}
+	clh.state.PID = pid
+
 	if err := clh.waitVMM(clhTimeout); err != nil {
 		clh.Logger().WithField("error", err).WithField("output", clh.cmdOutput.String()).Warn("cloud-hypervisor init failed")
 		clh.shutdownVirtiofsd()
 		return err
 	}
 
-	clh.state.PID = pid
+	if err := clh.bootVM(ctx); err != nil {
```

Comment:
One comment here. You chose to create the VMM and boot the VM in the same `startSandbox()` function, but you could have done it slightly differently too. The VMM can be started with only the API socket as argument (from `createSandbox()`), and later on, you can issue a `VmCreate(vmConfig)` to start the VM from `startSandbox()`.

Reply:
I think matching FC.go is okay for now.  We can look to see if there are any measurable benefits in a follow up.